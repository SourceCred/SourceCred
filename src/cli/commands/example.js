// @flow
import {Command, flags} from "@oclif/command";

export default class ExampleCommand extends Command {
  static description = "An example command description";
  static flags = {
    name: flags.string({char: "n", description: "name to print"}),
  };
  async run() {
    const {flags} = this.parse(ExampleCommand);
    const name = flags.name || "world";
    this.log(`hello ${name} EXAMPEL`);
  }
}
