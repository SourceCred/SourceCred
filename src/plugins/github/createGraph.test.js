// @flow

import {GraphView} from "./graphView";
import {exampleGraph} from "./example/example";

describe("plugins/github/createGraph", () => {
  it("example graph matches snapshot", () => {
    expect(exampleGraph()).toMatchSnapshot();
  });

  it("passes all GraphView invariants", () => {
    const view = new GraphView(exampleGraph());
    // This test is high leverage. It checks:
    // - that every node starting with a GitHub prefix
    //   - can be structured using fromRaw
    //   - has the correct type
    // - that every edge starting with a GitHub prefix
    //   - can be structured using fromRaw
    //   - and has the correct type,
    //   - and that its src has an expected prefix,
    //   - and that its dst has an expected prefix,
    // - that every child node
    //   - has exactly one parent
    //   - has a parent with the correct type
    view.checkInvariants();
    // as currently written, GV checks invariants on construction.
    // we call the method explicitly as a defensive step.
  });
});
