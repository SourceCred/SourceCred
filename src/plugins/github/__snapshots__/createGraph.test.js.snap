// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/github/createGraph example graph matches snapshot 1`] = `
Array [
  Object {
    "type": "sourcecred/graph",
    "version": "0.4.0",
  },
  Object {
    "edges": Array [
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "BOT",
          "credbot",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "417104047",
        ],
        "dstIndex": 13,
        "srcIndex": 30,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "1",
        ],
        "dstIndex": 17,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "10",
        ],
        "dstIndex": 18,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
        ],
        "dstIndex": 19,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "4",
        ],
        "dstIndex": 20,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
        ],
        "dstIndex": 21,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "7",
        ],
        "dstIndex": 22,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "8",
        ],
        "dstIndex": 23,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
        ],
        "dstIndex": 24,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
        ],
        "dstIndex": 25,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "9",
        ],
        "dstIndex": 26,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "373768703",
        ],
        "dstIndex": 2,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "373768850",
        ],
        "dstIndex": 3,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576185",
        ],
        "dstIndex": 4,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576220",
        ],
        "dstIndex": 5,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576248",
        ],
        "dstIndex": 6,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576273",
        ],
        "dstIndex": 7,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
        ],
        "dstIndex": 8,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576936",
        ],
        "dstIndex": 9,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "373768442",
        ],
        "dstIndex": 10,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "373768538",
        ],
        "dstIndex": 11,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "385223316",
        ],
        "dstIndex": 12,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "decentralion",
          "6",
          "COMMENT",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
          "369162222",
        ],
        "dstIndex": 14,
        "srcIndex": 31,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "10",
        ],
        "dstIndex": 18,
        "srcIndex": 32,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "9",
        ],
        "dstIndex": 26,
        "srcIndex": 32,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
          "5",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
        ],
        "dstIndex": 28,
        "srcIndex": 32,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
          "5",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100314038",
        ],
        "dstIndex": 29,
        "srcIndex": 32,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
          "6",
          "COMMENT",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
          "396430464",
        ],
        "dstIndex": 15,
        "srcIndex": 32,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "AUTHORS",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
          "7",
          "COMMENT",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
          "171460198",
        ],
        "dstIndex": 16,
        "srcIndex": 32,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "1",
        ],
        "dstIndex": 27,
        "srcIndex": 17,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "10",
        ],
        "dstIndex": 27,
        "srcIndex": 18,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
        ],
        "dstIndex": 27,
        "srcIndex": 19,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "4",
        ],
        "dstIndex": 27,
        "srcIndex": 20,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
        ],
        "dstIndex": 27,
        "srcIndex": 21,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "7",
        ],
        "dstIndex": 27,
        "srcIndex": 22,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "8",
        ],
        "dstIndex": 27,
        "srcIndex": 23,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
        ],
        "dstIndex": 27,
        "srcIndex": 24,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
        ],
        "dstIndex": 27,
        "srcIndex": 25,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "9",
        ],
        "dstIndex": 27,
        "srcIndex": 26,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "5",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
        ],
        "dstIndex": 25,
        "srcIndex": 28,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "5",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100314038",
        ],
        "dstIndex": 25,
        "srcIndex": 29,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "373768703",
        ],
        "dstIndex": 19,
        "srcIndex": 2,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "373768850",
        ],
        "dstIndex": 19,
        "srcIndex": 3,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576185",
        ],
        "dstIndex": 19,
        "srcIndex": 4,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576220",
        ],
        "dstIndex": 19,
        "srcIndex": 5,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576248",
        ],
        "dstIndex": 19,
        "srcIndex": 6,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576273",
        ],
        "dstIndex": 19,
        "srcIndex": 7,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
        ],
        "dstIndex": 19,
        "srcIndex": 8,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576936",
        ],
        "dstIndex": 19,
        "srcIndex": 9,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "373768442",
        ],
        "dstIndex": 21,
        "srcIndex": 10,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "373768538",
        ],
        "dstIndex": 21,
        "srcIndex": 11,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "385223316",
        ],
        "dstIndex": 21,
        "srcIndex": 12,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "417104047",
        ],
        "dstIndex": 21,
        "srcIndex": 13,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
          "369162222",
        ],
        "dstIndex": 24,
        "srcIndex": 14,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "6",
          "COMMENT",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
          "396430464",
        ],
        "dstIndex": 25,
        "srcIndex": 15,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "HAS_PARENT",
          "7",
          "COMMENT",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
          "171460198",
        ],
        "dstIndex": 28,
        "srcIndex": 16,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "MENTIONS_AUTHOR",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
          "6",
          "COMMENT",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
          "396430464",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
        ],
        "dstIndex": 15,
        "srcIndex": 25,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "MERGED_AS",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
        ],
        "dstIndex": 0,
        "srcIndex": 24,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "MERGED_AS",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
        ],
        "dstIndex": 1,
        "srcIndex": 25,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "10",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "10",
        ],
        "dstIndex": 18,
        "srcIndex": 18,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "10",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
        ],
        "dstIndex": 19,
        "srcIndex": 18,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "1",
        ],
        "dstIndex": 17,
        "srcIndex": 19,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
        ],
        "dstIndex": 32,
        "srcIndex": 25,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "373768703",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
        ],
        "dstIndex": 21,
        "srcIndex": 2,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "373768850",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "373768538",
        ],
        "dstIndex": 11,
        "srcIndex": 3,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576185",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "5",
        ],
        "dstIndex": 25,
        "srcIndex": 4,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576220",
          "5",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
        ],
        "dstIndex": 28,
        "srcIndex": 5,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576248",
          "7",
          "COMMENT",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
          "171460198",
        ],
        "dstIndex": 16,
        "srcIndex": 6,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576273",
          "3",
          "USERLIKE",
          "USER",
          "wchargin",
        ],
        "dstIndex": 32,
        "srcIndex": 7,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "1",
        ],
        "dstIndex": 17,
        "srcIndex": 8,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
        ],
        "dstIndex": 19,
        "srcIndex": 8,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
          "4",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
        ],
        "dstIndex": 24,
        "srcIndex": 8,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
          "5",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
        ],
        "dstIndex": 28,
        "srcIndex": 8,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
          "385576920",
          "7",
          "COMMENT",
          "REVIEW",
          "sourcecred",
          "example-github",
          "5",
          "100313899",
          "171460198",
        ],
        "dstIndex": 16,
        "srcIndex": 8,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "ISSUE",
          "sourcecred",
          "example-github",
          "6",
          "385223316",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
        ],
        "dstIndex": 19,
        "srcIndex": 12,
      },
      Object {
        "address": Array [
          "sourcecred",
          "github",
          "REFERENCES",
          "6",
          "COMMENT",
          "PULL",
          "sourcecred",
          "example-github",
          "3",
          "369162222",
          "4",
          "ISSUE",
          "sourcecred",
          "example-github",
          "2",
        ],
        "dstIndex": 19,
        "srcIndex": 14,
      },
    ],
    "nodes": Array [
      Array [
        "sourcecred",
        "git",
        "COMMIT",
        "0a223346b4e6dec0127b1e6aa892c4ee0424b66a",
      ],
      Array [
        "sourcecred",
        "git",
        "COMMIT",
        "6d5b3aa31ebb68a06ceb46bbd6cf49b6ccd6f5e6",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "373768703",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "373768850",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "385576185",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "385576220",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "385576248",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "385576273",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "385576920",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
        "385576936",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "6",
        "373768442",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "6",
        "373768538",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "6",
        "385223316",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "ISSUE",
        "sourcecred",
        "example-github",
        "6",
        "417104047",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "PULL",
        "sourcecred",
        "example-github",
        "3",
        "369162222",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "PULL",
        "sourcecred",
        "example-github",
        "5",
        "396430464",
      ],
      Array [
        "sourcecred",
        "github",
        "COMMENT",
        "REVIEW",
        "sourcecred",
        "example-github",
        "5",
        "100313899",
        "171460198",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "1",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "10",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "2",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "4",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "6",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "7",
      ],
      Array [
        "sourcecred",
        "github",
        "ISSUE",
        "sourcecred",
        "example-github",
        "8",
      ],
      Array [
        "sourcecred",
        "github",
        "PULL",
        "sourcecred",
        "example-github",
        "3",
      ],
      Array [
        "sourcecred",
        "github",
        "PULL",
        "sourcecred",
        "example-github",
        "5",
      ],
      Array [
        "sourcecred",
        "github",
        "PULL",
        "sourcecred",
        "example-github",
        "9",
      ],
      Array [
        "sourcecred",
        "github",
        "REPO",
        "sourcecred",
        "example-github",
      ],
      Array [
        "sourcecred",
        "github",
        "REVIEW",
        "sourcecred",
        "example-github",
        "5",
        "100313899",
      ],
      Array [
        "sourcecred",
        "github",
        "REVIEW",
        "sourcecred",
        "example-github",
        "5",
        "100314038",
      ],
      Array [
        "sourcecred",
        "github",
        "USERLIKE",
        "BOT",
        "credbot",
      ],
      Array [
        "sourcecred",
        "github",
        "USERLIKE",
        "USER",
        "decentralion",
      ],
      Array [
        "sourcecred",
        "github",
        "USERLIKE",
        "USER",
        "wchargin",
      ],
    ],
  },
]
`;
